src/main/java/frc/robot/commands/BalancerCommand.java

```diff
@@ -36,28 +41,46 @@ public void initialize() {
     // Called every time the scheduler runs while the command is scheduled.
     @Override
     public void execute() {
+        // currentAngle = ahrs.getPitch();
         if(!reachedStation) {
-            returnPower = 0.90;
+            returnPower = 0.80;
             System.out.println(ahrs.getPitch());
             if(ahrs.getPitch() >= 15.0) reachedStation = true;
         }
         else{
-            returnPower = -1 * pid.calculate(ahrs.getPitch(), 0);
-            // returnPower  = 1.8 * 0.01142857143 * ahrs.getPitch(); // MAX_ERR * Kp = MAX_PWR   35 * Kp = 0.4  Kp = 0.4 / 35
-            if(!timerEnabled && Math.abs(ahrs.getPitch()) <= 2.0){
-                stoptimer.stop();
-                stoptimer.reset();
-                timerEnabled = true;
+            if(!passedCenter){
+                returnPower = 0.3;
+                if(ahrs.getPitch() <= -5.0) passedCenter = true;
             }
-            if(Math.abs(ahrs.getPitch()) >= 2.0){
-                stoptimer.stop();
-                stoptimer.reset();
-                timerEnabled = false;
+            else{
+                returnPower = -0.25;
+                if(Math.abs(ahrs.getPitch()) <= 2.0){
+                    returnPower = 0.0;
+                    driveSubsystem.lockNow();
```

Comment:
This doesn't require `BalancerCommand` to only work with a `BaseSwerveSubsystem`, because you could do
```java
if (driveSubsystem instanceof BaseSwerveSubsystem) ((BaseSwerveSubsystem) driveSubsystem).lockNow();
```

Reply:
As long as the drive forward logic is abstracted, this command can still work with both tank and swerve.